source/app/views/urls/index.html.erb

```diff
@@ -0,0 +1,14 @@
+<p>Shorten URL:</p>
+
+<%= flash[:error] if !nil? %>
+
+<%= form_for :url do |f| %>
+  <%= f.label :long_url %>
+  <%= f.text_field :long_url %>
+<% end %>
+
+<ul>
+<% @urls.each do |url| %>
+  <li><%= "0.0.0.0:3000/urls/#{url.id} ====> #{url.long_url} (clicked #{url.click_count} times)" %></li>
```

Comment:
Your routes file has a `:urls` resource with a `:show` actions, so it should generate route helper methods for you. The name of it is kinda unfortunate in this case:

I think `url_url(url)` should output something like `"0.0.0.0:3000/urls/#{url.id}"` here. (and will remain correct in non-development environments.